src/gui/update.rs

```diff
@@ -70,21 +70,29 @@ pub fn handle_message(ajour: &mut Ajour, message: Message) -> Result<Command<Mes
                 ));
             }
 
+            // NOTE (casperstorm): Migration from single World of Warcraft directory to multiple directories.
             let flavors = &Flavor::ALL[..];
+            if ajour.config.wow.directory.is_some() {
+                for flavor in flavors {
+                    let path = ajour.config.wow.directory.as_ref().unwrap();
+                    let flavor_path = ajour.config.get_flavor_directory_for_flavor(flavor, path);
+                    if flavor_path.exists() {
+                        ajour.config.wow.directories.insert(*flavor, flavor_path);
+                    }
+                }
+
+                ajour.config.wow.directory = None;
+                let _ = &ajour.config.save();
+            }
+
+            let flavors = ajour.config.wow.directories.keys().collect::<Vec<_>>();
```

Comment:
Should we move this to our `apply_config` function in `src/gui` since its run only once, immediately on startup before the GUI loads the first time?

Reply:
Good point, ill look at that.

Reply:
Great! And as we discussed, probably worth keeping some logic in the Message::Parse where it applies any found new flavor folders to unmapped flavor directories, in instances where people install classic to the existing wow directory or something

Reply:
I moved it to `apply_config` - how do you like it there? https://github.com/ajour/ajour/pull/574/commits/2285ed23d91203b01522ba9e3b74b67be0dcf62f